src/AutoMapper/Execution/ReflectionHelper.cs

```diff
@@ -22,7 +22,7 @@ public static object GetDefaultValue(this ParameterInfo parameter)
         public static object MapMember(this ResolutionContext context, MemberInfo member, object value)
         {
             var memberType = member.GetMemberType();
-            return context.Map(value, null, value?.GetType() ?? memberType, memberType);
+            return context.Mapper.Map(value, null, value?.GetType() ?? memberType, memberType, context);

```

Comment:
Here the idea was to reuse the context. This doesn't. If you can make it a struct it won't matter anymore.